src/MessageManager.cpp

```diff
@@ -207,7 +215,7 @@ void MessageManager::Unsubscribe( IMessageSubscriber* pSubscriber, MessageID m )
 void MessageManager::Broadcast( Message &msg ) const
 {
 	// GAMESTATE is created before MESSAGEMAN, and has several BroadcastOnChangePtr members, so they all broadcast when they're initialized.
-	if(m_Logging)
+	if(this != nullptr && m_Logging)
```

Comment:
This discussion again? We'll just have to clean up our code even more.

For new developers, the warning that results with this code is the following:

```
'this' pointer cannot be null in well-defined C++ code;
comparison may be assumed to always evaluate to true
```

The fact that we (allegedly) need this is a sign that we have improvements to make. Can we at least confirm if this is still necessary?